testutil/network/network.go

```diff
@@ -165,15 +166,86 @@ type (
 	}
 )
 
+type TestnetEnv interface {
+	Log(args ...interface{})
+	Logf(format string, args ...interface{})
+	ConfigDir(chainID string) (string, error)
+	AssertNoError(err error)
+}
+
+var _ TestnetEnv = (*TestingT)(nil)
+
+type TestingT struct {
+	*testing.T
+}
+
+func (t TestingT) AssertNoError(err error) {
+	require.NoError(t.T, err)
+}
+
+func (t TestingT) Log(args ...interface{}) {
+	t.T.Log(args...)
+}
+
+func (t TestingT) Logf(format string, args ...interface{}) {
+	t.T.Logf(format, args...)
+}
+
+func (t TestingT) ConfigDir(chainID string) (string, error) {
+	return ioutil.TempDir(t.TempDir(), chainID)
+}
+
+var _ TestnetEnv = (*StandaloneTestnet)(nil)
+
+type StandaloneTestnet struct {
+	cmd           *cobra.Command
+	baseConfigDir string
+}
+
+func (s StandaloneTestnet) Log(args ...interface{}) {
+	s.cmd.Println(args...)
+}
+
+func (s StandaloneTestnet) Logf(format string, args ...interface{}) {
+	s.cmd.Printf(format, args...)
+}
+
+func (s StandaloneTestnet) AssertNoError(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (s StandaloneTestnet) ConfigDir(chainID string) (string, error) {
+	configDir := fmt.Sprintf("%s/%s", s.baseConfigDir, chainID)
+	if _, err := os.Stat(configDir); !os.IsNotExist(err) {
+		return "", fmt.Errorf("config directory already exists: %s", configDir)
+	}
+	return configDir, nil
+}
+
+func NewStandaloneTestnetEnv(cmd *cobra.Command, configDir string) StandaloneTestnet {
+	return StandaloneTestnet{
+		cmd,
+		configDir,
+	}
+}
+
+func NewForTesting(t *testing.T, cfg Config) *Network {
+	testnetEnv := TestingT{t}
+	return New(testnetEnv, cfg)
+
+}
+
 // New creates a new Network for integration tests.
-func New(t *testing.T, cfg Config) *Network {
+func New(t TestnetEnv, cfg Config) *Network {
 	// only one caller/test can create and use a network at a time
 	t.Log("acquiring test network lock")
 	lock.Lock()
 
-	baseDir, err := ioutil.TempDir(t.TempDir(), cfg.ChainID)
-	require.NoError(t, err)
-	t.Logf("created temporary directory: %s", baseDir)
+	baseDir, err := t.ConfigDir(cfg.ChainID)
```

Comment:
maybe `baseDir` can be an arg to `New`, so that we remove `t.ConfigDir`? 

we could probably remove `NewForTesting` too then, and just do `s.network = network.New(s.T(), t.TempDir(), cfg)` in tests.